### PR TITLE
nalu-nalu 

### PR DESCRIPTION
**Pull request by michaeljbrazell**
- added ability to run nalu-nalu from exawind-driver
- added nalu-nalu-cylinder case from old python driver  
- remove all amr-wind specific inputs and driver should only create nalu instances
- nalu instances (for now) are assumed exclusive and equally load balanced

### CODE REVIEW

**Comment by michaeljbrazell** [comment]
I don't think the new test works yet... maybe there is an issue with mesh_tag_offset?

**Comment by michaeljbrazell** [comment]
I don't think the new test works yet... maybe there is an issue with mesh_tag_offset?

**Comment by marchdf** [comment]
Should we merge #16 ?

**Comment by michaeljbrazell** [comment]
it's not merged in nalu-wind yet...

**Comment by jrood-nrel** [comment]
It is now

**Comment by michaeljbrazell** [comment]
thanks @jrood-nrel ! back to #16, I can't remember if we figured out the timers? otherwise I'm ok with it being merged

**Comment by marchdf** [comment]
we need to figure out the timers....

**Comment by michaeljbrazell** [comment]
well I'm stumped. the iblanks look good for the cylinder but the solution is not being exchanged. Also, when I turn off mesh movement on the cylinder the driver crashes... which is suspicious. There is a chance this can be fixed through input files but I don't know nalu-wind well enough to know for sure

**Comment by michaeljbrazell** [comment]
runs now with mesh movement off, nalu-wind expected wall velocities from mesh movement and that caused a crash. Set wall velocities to [0,0,0]

**Comment by michaeljbrazell** [comment]
just need to figure out the added TODO then should be good to merge

File: src/OversetSimulation.cpp [hunk 1]
```diff
@@ -103,14 +103,17 @@ void OversetSimulation::exchange_solution()
     if (m_has_amr) {
         m_tg.dataUpdate_AMR();
     } else {
-        throw std::runtime_error("Invalid overset exchange");
+        const int row = 0;
+        // TODO: need a way to count nodal variable in nalu-wind, right now hard-coded to 4
```
Comment (psakievich) [inline src/OversetSimulation.cpp:107]: I think this is what you want: https://github.com/Exawind/nalu-wind/blob/4d212ec3991ab2ddc9d395f2547442dd73863fe0/src/overset/ExtOverset.C#L161-L180

Comment (michaeljbrazell) [reply src/OversetSimulation.cpp:107]: yep, that'll do it!

**Comment by michaeljbrazell** [comment]
if folks are ok with the assumption using m_solvers[0] this should be ready to merge

File: src/OversetSimulation.cpp [hunk 2]
```diff
@@ -103,8 +103,12 @@ void OversetSimulation::exchange_solution()
     if (m_has_amr) {
         m_tg.dataUpdate_AMR();
     } else {
-        throw std::runtime_error("Invalid overset exchange");
+        const int row_major = 0;
+        // assuming this pathway is nalu-wind only and all instances have same number of field components
+        const int ncomps = m_solvers[0]->get_ncomps();
+        m_tg.dataUpdate(ncomps, row_major);
     }
+
```
Comment (marchdf) [inline src/OversetSimulation.cpp:111]: remove spurious new line?

File: src/OversetSimulation.cpp [hunk 3]
```diff
@@ -103,8 +103,12 @@ void OversetSimulation::exchange_solution()
     if (m_has_amr) {
         m_tg.dataUpdate_AMR();
     } else {
-        throw std::runtime_error("Invalid overset exchange");
+        const int row_major = 0;
```
Comment (marchdf) [inline src/OversetSimulation.cpp:106]: do we want `else if nalu-wind only and the else original error?

Comment (michaeljbrazell) [reply src/OversetSimulation.cpp:106]: I think we have an assumption and a throw elsewhere that guarantees at least one solver is unstructured (nalu-wind)

File: app/exawind/exawind.cpp [hunk 4]
```diff
@@ -118,11 +132,12 @@ int main(int argc, char** argv)
     }
 
     exawind::OversetSimulation sim(MPI_COMM_WORLD);
-    sim.echo(
-        "Initializing AMR-Wind on " + std::to_string(num_awind_ranks) +
-        " MPI ranks");
-    if (amr_comm != MPI_COMM_NULL)
+    if (amr_comm != MPI_COMM_NULL) {
```
Comment (marchdf) [inline app/exawind/exawind.cpp:135]: starting here you start checking `amr_comm` instead of `use_amr_wind`. Want to just use `use_amr_wind` check?

Comment (michaeljbrazell) [reply app/exawind/exawind.cpp:135]: yep I can do that!

**Review by marchdf — APPROVED** [review]

**Review by psakievich — APPROVED** [review]

**Review by jrood-nrel — APPROVED** [review]

**Pull request merged**